### PR TITLE
Adding ( [ONNX] Extend ONNX Frontend with Function Mish-18 )

### LINKED ISSUE #20550: [Good First Issue]: Extend ONNX Frontend with Function Mish-18
### Context

Neural networks are graphs consisting of nodes called operators. Each operator corresponds to a mathematical function, usually described in framework's documentation or an AI standard, such as ONNX.
OpenVINO ONNX Frontend is a component responsible for working with ONNX graphs and requires implementation of different ONNX operators in order to use ONNX models.
This task requires extending OpenVINO ONNX Frontend with Function Mish.
Necessary help will be provided by ONNX Fronted team.

### What needs to be done?

Operator details can be found in [ONNX Operators](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Mish)
More details can be found in [ONNX Changelog](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Mish-18)
1. Create `.hpp` and `.cpp` files [here](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/frontend/src/op)
2. Prepare an implementation of this operator in form of a function. Mish-18 should be placed in [opset 1 namespace](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/op/unique.hpp#L13).
3. Register the function in [ops_bridge.cpp](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/frontend/src/ops_bridge.cpp#L554-L560) while keeping alphabetical order
4. Create test model(s) in [ONNX models directory](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/onnx/tests/models/). OpenVINO test infrastructure then converts prototxt files to ONNX models - you will use those models later in tests
5. Add tests covering all use cases [here](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/onnx_import.in.cpp)
6. Check [Python xfailed tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/tests/tests_python/test_backend.py) to find a test marked as a xfailed for added functionality. If any exist - remove corresponding lines and try to verify by using cmdline "python -m pytest -k name_of_test".
More details in [adding operators to ONNX Frontend guide](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/how_to_add_op.md)

### Example Pull Requests

- https://github.com/openvinotoolkit/openvino/pull/15076
- https://github.com/openvinotoolkit/openvino/pull/14936
- https://github.com/openvinotoolkit/openvino/pull/12510
- https://github.com/openvinotoolkit/openvino/pull/19428

### Resources

- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)
- Guides on [ONNX Frontend architecture](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/architecture.md) and [ONNX Frontend tests](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/onnx/docs/tests.md) are also available
- You can also see a [blog post on contributing to OpenVINO](https://medium.com/openvino-toolkit/how-to-contribute-to-an-ai-open-source-project-c741f48e009e)

### Contact points

@gkrivor

### Ticket

_No response_

### PR DESCRIPTION
**Pull request by PRATHAM-SPS**
### Details:
 - *Expand the ONNX Frontend Capabilities by Incorporating the Mish-18 Function*

### Tickets:
 - *Closes #20550*

### CODE REVIEW

File: src/frontends/onnx/frontend/src/ops_bridge.cpp [hunk 1]
```diff
@@ -115,6 +115,7 @@
 #include "op/min.hpp"
 #include "op/mod.hpp"
 #include "op/mul.hpp"
+#include "op/mish.hpp"
```
Comment (mitruska) [inline src/frontends/onnx/frontend/src/ops_bridge.cpp:118]: It should keep alphabetical order (required by style check):
```suggestion
#include "op/mish.hpp"
#include "op/mod.hpp"
#include "op/mul.hpp"
```

**Review by mitruska — COMMENTED** [review]
Hello @PRATHAM-SPS, thank you for this contribution! 
In general the import code looks correct, just few comments.
Also please align the code style to pass the [CI check](https://github.com/openvinotoolkit/openvino/actions/runs/7900061439/job/21562522084?pr=22833),
call `make clang_format_fix_all` from build directory to apply auto formatting.

**Comment by PRATHAM-SPS** [comment]
> Hello @PRATHAM-SPS, thank you for this contribution! In general the import code looks correct, just few comments. Also please align the code style to pass the [CI check](https://github.com/openvinotoolkit/openvino/actions/runs/7900061439/job/21562522084?pr=22833), call `make clang_format_fix_all` from build directory to apply auto formatting.

image
I am facing this issue , I have also tried building the cmake with "cmake -DENABLE_PYTHON=ON -DENABLE_TESTS=ON -DENABLE_PROFILING_ITT=ON -DSELECTIVE_BUILD=COLLECT -B build"

Any suggestion?? How should I fix it?

File: src/frontends/onnx/frontend/src/op/mish.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace ov {
+namespace frontend {
+namespace onnx {
+namespace op {
+namespace set_1 {
+ov::OutputVector mish(const ov::frontend::onnx::Node& node) {
+    const auto data = node.get_ov_inputs().at(0);
+
+    return std::make_shared<v4::Mish>(data)->outputs();
```
[the remark this answers is not in the document]
Comment (PRATHAM-SPS) [reply src/frontends/onnx/frontend/src/op/mish.cpp]: No there is no specific reason for that, I have commited all the changes that you have suggested in the new commit.Thank you for your guidance and support .

**Comment by PRATHAM-SPS** [comment]
> > > Hello @PRATHAM-SPS, thank you for this contribution! In general the import code looks correct, just few comments. Also please align the code style to pass the [CI check](https://github.com/openvinotoolkit/openvino/actions/runs/7900061439/job/21562522084?pr=22833), call `make clang_format_fix_all` from build directory to apply auto formatting.
> > 
> > 
> > image I am facing this issue , I have also tried building the cmake with "cmake -DENABLE_PYTHON=ON -DENABLE_TESTS=ON -DENABLE_PROFILING_ITT=ON -DSELECTIVE_BUILD=COLLECT -B build"
> > Any suggestion?? How should I fix it?
> 
> you need install clang-format-9 tool

Is there any other solution for it? I have ubuntu 22.04 it come with clang-format-14 and I am facing lots of issues to install the clang-format-9 externally.

File: src/frontends/onnx/tests/onnx_import.in.cpp [hunk 3]
```diff
@@ -6245,3 +6245,18 @@ OPENVINO_TEST(${BACKEND_NAME}, onnx_model_gelu_float_tanh) {
...
+    auto test_case = ov::test::TestCase(model, s_device);
+    test_case.add_input<float>({ 1.8079f, -0.2892f, 2.0915f,  12.5101f, -1.8837f, 0.2586f, 2.9528f,  0.001f,
+                            6.0296f, -1.0745f, -0.2703f, 1.319f,   -3.3607f, 0.1434f, -8.4590f, 0.0f,
+                            2.7608f, 0.3126f,  0.3f,     3.0f,     7.6919f,  0.5859f, -11.992f, -37.8f });
+
+    test_case.add_expected_output<float>({1.737521f, -0.146684f, 2.041557f,  12.5101f,  -0.264820f, 0.176079f, 2.938304f,  0.0006f,
+                            6.029531f, -0.306873f, -0.138725f, 1.206575f, -0.114629f, 0.092553f, -0.001792f, 0.0f,
+                            2.741334f, 0.217909f,  0.208001f,  2.986535f, 7.691896f,  0.453058f, -0.000074f, 0.0f });
+
+   test_case.run_with_tolerance_as_fp(0.000001f);
```
Comment (mitruska) [inline src/frontends/onnx/tests/onnx_import.in.cpp:6261]: code style:
```suggestion
    test_case.add_input<float>({1.8079f, -0.2892f, 2.0915f,  12.5101f, -1.8837f, 0.2586f, 2.9528f,  0.001f,
                                6.0296f, -1.0745f, -0.2703f, 1.319f,   -3.3607f, 0.1434f, -8.4590f, 0.0f,
                                2.7608f, 0.3126f,  0.3f,     3.0f,     7.6919f,  0.5859f, -11.992f, -37.8f});

    test_case.add_expected_output<float>({1.737521f,  -0.146684f, 2.041557f,  12.5101f,   -0.264820f, 0.176079f,
                                          2.938304f,  0.0006f,    6.029531f,  -0.306873f, -0.138725f, 1.206575f,
                                          -0.114629f, 0.092553f,  -0.001792f, 0.0f,       2.741334f,  0.217909f,
                                          0.208001f,  2.986535f,  7.691896f,  0.453058f,  -0.000074f, 0.0f});

    test_case.run_with_tolerance_as_fp(0.000001f);
```

**Review by mitruska — COMMENTED** [review]
@PRATHAM-SPS You can also try to format the files with your IDE, there is a chance that the output will be compatible. If not, I've added the code style suggestions in the comments, that should fix the style-check.

**Comment by p-wysocki** [comment]
> Is there any other solution for it? I have ubuntu 22.04 it come with clang-format-14 and I am facing lots of issues to install the clang-format-9 externally.

Hi @PRATHAM-SPS, we started a discussion on that internally - we'll do our best to provide a solution to that as soon as possible. :) For now please use @mitruska's comments.

**Comment by PRATHAM-SPS** [comment]
Hello @mitruska @p-wysocki @ilya-lavrenov  can someone please initiate CI checks ?

**Comment by mitruska** [comment]
build_jenkins

**Review by mitruska — APPROVED** [review]

**Review by mmikolajcz — APPROVED** [review]

**Pull request merged**